### PR TITLE
add new-rule-format transform and initial repo structure

### PR DESCRIPTION
**Pull request by vitorbal**
I ran the transform on our own core rules one more time just to make sure everything was fine, and I also added some tests here.

Another thing to note is that I found a couple of eslint-plugins in the wild that use ES6 module syntax for their rules, which the transform does not support currently.

### CODE REVIEW

**Comment by ilyavolodin** [comment]
Maybe it's worth mentioning in the Readme file that those codemods target ES5 rules, and will not work for ES6? Other than that, looks good to me, thanks!

**Comment by nzakas** [comment]
Agreed. If there are some rules in the wild that are using ES6, we should make it clear that this only works on ES5 rules.

**Comment by vitorbal** [comment]
@ilyavolodin @nzakas makes sense! I'll add a note.

I'm working on making the transform run for ES6 rules as well, I'll try to sneak that in before the announcement.

**Comment by vitorbal** [comment]
Added a note.

**Comment by ilyavolodin** [comment]
LGTM

**Comment by nzakas** [comment]
Ooh, I just realized something: if we don't publish this to npm, then someone could grab this name. Maybe a simple CLI would be good to include? (Also helps with versioning.)

**Comment by vitorbal** [comment]
I pushed a simple CLI, added eslint-release for publishing and updated the README. Let me know what you think!

File: README.md [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+
+```
+$ eslint-transforms <transform-name> <path>
+```
+
+or
+
+```
+$ npm run <transform-name> -- <path>
+```
```
Comment (nzakas) [inline README.md]: I don't think this will work when installed from npm.

**Comment by nzakas** [comment]
Looks like a good start! Can I ask a favor? We try to keep all the repos using the same basic setup and tools so people can easily move between repos to make changes. I noticed you're using ES6 and Jest, which is very different from our other projects. Would you mind switching to our standard ES5 and Mocha/Chai setup?

Also, we keep tests in a top level directory called `tests` and test fixtures in `tests/fixtures`, would you mind moving the tests and fixtures?

**Comment by vitorbal** [comment]
@nzakad Yes, absolutely! I will work on it this week. Sorry for the delayed
response, i was out of the country for the weekend :)

The only thing i wonder is if migrating away from Jest will be a big
hassle, since I'm using the test utils provided by jscodeshift to test
transforms, and they were designed to work with Jest. I'll have to check if
we can plug it into a Mocha/Chai setup

**Comment by vitorbal** [comment]
@ilyavolodin @nzakas Refactored everything to use ES5 only, and changed the tests to use Mocha/Chai instead of Jest. Let me know what you think!

File: lib/new-rule-format/new-rule-format.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+ * module.exports = {
+ *     meta: {
+ *         docs: {},
+ *         schema: []
+ *     },
+ *     create: function(context) { ... }
+ * };
+ * ```
+ * @author Vitor Balocco
+ * @copyright jQuery Foundation and other contributors, https://jquery.org/
```
Comment (nzakas) [inline lib/new-rule-format/new-rule-format.js]: We don't need copyright or license here.

File: lib/new-rule-format/new-rule-format.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    return exportsExpression;
+}
+
+//------------------------------------------------------------------------------
+// Transform Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @param {Object} fileInfo - holds information about the currently processed file.
+ * @param {Object} api - exposes the jscodeshift api.
```
Comment (nzakas) [inline lib/new-rule-format/new-rule-format.js]: I don't see this argument anywhere. Safe to remove?

**Comment by nzakas** [comment]
Thanks! Just noticed a couple minor docs things. Otherwise, lgtm.

File: lib/new-rule-format/new-rule-format.js (hunk 3 above)
Comment (vitorbal) [inline lib/new-rule-format/new-rule-format.js]: Oops, you are totally right, it's gone. Thanks!

File: lib/new-rule-format/new-rule-format.js (hunk 2 above)
Comment (vitorbal) [inline lib/new-rule-format/new-rule-format.js]: Alright, removed it!

**Comment by vitorbal** [comment]
Thanks again for the feedback! Took care of everything in the latest commit.

**Comment by nzakas** [comment]
Great, I think this is a strong starting point. We can handle anything else that comes up as bugs.

**Pull request merged**